doc/querying.md

```diff
@@ -0,0 +1,240 @@
+```
+
+### extend-with-aggregate
+
+COUNT example:
+
+```clojure
+(require '[com.verybigthings.penkala.helpers :as h])
+
+(-> users-rel
+  (r/extend-with-aggregate :count [:count (h/l 1)])
+  (r/get-select-query {}))
+
+=> ["SELECT count(1) AS \"count\", \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" GROUP BY \"users\".\"id\", \"users\".\"is_admin\", \"users\".\"username\""]
+```
+
+### having
+
+And having operation. If there's already a having clause set, this clause will be joined with AND.
+
+```clojure
+(-> users-rel
+  (r/extend-with-aggregate :count [:count (h/l 1)])
+  (r/having [:> :count (h/l 1)])
+  (r/get-select-query {}))
+
+=> ["SELECT count(1) AS \"count\", \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" GROUP BY \"users\".\"id\", \"users\".\"is_admin\", \"users\".\"username\" HAVING count(1) > 1"]
+```
+
+### intersect
+
+Creates a relation that is a combination of two relations with the INTERSECT operator.
+
+```clojure
+;; ignored by seancorfield/readme
+
+(let [users (:users *env*)]
+  (select! *env* (r/intersect users (r/where users [:is-true :is-admin]))))
+
+=> [#:users{:username test@test.com, :is_admin false, :id 1}]
+```
+
+```clojure
+(-> users-rel
+	(r/intersect admins-rel)
+	(r/get-select-query {}))
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is-admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM (SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" INTERSECT SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" WHERE \"users\".\"is_admin\" IS TRUE) AS \"users\""]
+```
+
+### join
+
+```clojure
+(def posts-spec {:name "posts"
+                 :columns ["id" "user_id" "body"]
+                 :pk ["id"]})
+  
+(def posts-rel (r/spec->relation users-spec))
+
+(r/get-select-query (r/join posts-rel :left admins-rel :author [:= :user-id :author/id]) {})
+```
+
+### limit
+
+```clojure
+(r/get-select-query (r/limit users-rel 10) {})
+```
+
+### lock
+
+Locks selected rows.
+
+```clojure
+;; ignored by seancorfield/readme
+
+(select! *env* (-> *env* :users (r/lock :share)))
+
+=> [#:users{:is-admin false, :username test@test.com, :id 1} #:users{:is-admin true, :username admin@test.com, :id 2}]
+```
+
+```clojure
+(r/get-select-query (r/lock users-rel :share) {})
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" FOR SHARE"]
+```
+
+### offset
+
+```clojure
+(r/get-select-query (r/offset users-rel 10) {})
+```
+
+### order-by
```

Comment:
Can we order by multiple columns? If yes, how?

Reply:
Of course, I'll add.